src/Notify.Api.Factory.pas

```diff
@@ -0,0 +1,37 @@
+unit Notify.Api.Factory;
+
+interface
+
+uses
+  Notify.Api.Contract;
+
+type
+  TNotifyApiFactory = class(TInterfacedObject, INotifyApiFactory)
+  public
+    class function New: INotifyApiFactory;
+    function Api: INotifyApi;
+  end;
+
+implementation
+
+uses
+  Notify.Api.NetHTTP,
+  Notify.Api.Indy;
+
+{ TNotifyApiFactory }
+
+function TNotifyApiFactory.Api: INotifyApi;
+begin
+  {$IFDEF INDY}
```

Comment:
This needs documenting in the README.md. It's effectively a part of your public interface "how to switch between Indy/NetHTTP".

Also, consider renaming this compiler define to `NTFY_INDY`. Consider, with such a generic define based on the name of a very popular network component set, what would happen if your library was included in other large projects that may have libraries with the same defines for other purposes? You may even consider having a more descriptive name `NTFY_DO_A_THING_WITH_INDY` so it becomes self-documenting when people include it in their projects.

Reply:
Marking out for improvement. I'll make the necessary changes. I think `NTFY_INDY` is more descriptive name as you mentioned as it avoids clashing with other name that a large library might have. I'll also provide documentation of that in the README.md file.

Reply:
@rhatherall, your review has been submitted. I will really appreciate if you have any other suggestions.